esmvalcore/_recipe.py

```diff
@@ -488,13 +510,66 @@ def _update_fx_settings(settings, variable, config_user):
         settings['weighting_landsea_fraction']['fx_files'] = fx_dict
         logger.info(msg, 'land/sea fraction weighting', pformat(fx_dict))
 
-    for step in ('area_statistics', 'volume_statistics'):
+    for step in ('area_statistics', 'volume_statistics', 'zonal_statistics'):
+        var = dict(variable)
+        cmor_fxvars = []
+        fx_files_dict = {}
         if settings.get(step, {}).get('fx_files'):
-            var = dict(variable)
             var['fx_files'] = settings.get(step, {}).get('fx_files')
-            fx_files_dict = {
-                fxvar: _get_correct_fx_file(variable, fxvar, config_user)
-                for fxvar in var['fx_files']}
+            for fxvar in var['fx_files']:
+                fxvar_name = fxvar
+                if isinstance(fxvar, dict):
+                    fxvar_name = fxvar.get("short_name")
+                    if not fxvar_name:
+                        raise RecipeError(
+                            "Key short_name missing from {}".format(fxvar))
+                fx_file, cmor_fx_var = _get_correct_fx_file(
+                    variable, fxvar, config_user)
+                fx_files_dict[fxvar_name] = fx_file
+                if cmor_fx_var:
+                    cmor_fxvars.append(cmor_fx_var)
+
+            # special case for preprocessing fx variables
+            if var.get("fx_var_preprocess"):
```

Comment:
I need that to be set as a special case because of this line: l.753 `if not ancestors or variable.get("fx_var_preprocess"):` -> this is a special case that allows the code to get the files for the main variable even if there are ancestors. If we set that True all the time it'll wreak havoc when there are no fx_files preprocs